libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/plugin/mpp/CompilationDetails.kt

```diff
@@ -96,6 +96,7 @@ open class DefaultCompilationDetails<T : KotlinCommonOptions>(
             lowerCamelCaseName(
                 target.disambiguationClassifier,
                 compilationPurpose.takeIf { it != KotlinCompilation.MAIN_COMPILATION_NAME },
+                "compilation",
```

Comment:
nit: I'd suggest using "classpath" or "MppClasspath" suffix. Otherwise, "jvmMainCompilationRuntimeOnly" looks a bit strange IMHO. If the goal is not to make them user-facing, any name will do :)

Also, do you know what is the difference between these configurations and "runtimeClasspath"/"compileClasspath" ones?

Reply:
Good point! This are effectively the platform specific `runtimeClasspath` / `compileClasspath` configuratoins, 
So a name like `jvmRuntimeClasspath` or `jvmCompileClasspath` are more idiomatic namings.

Reply:
I think I was wrong with my comment. There are configurations like `runtimeClasspath` and `compileClasspath` for platforms, however those configurations here are used to build those. 

This here is the compilations equivalent of the source sets `implementation`, `api`, ... configurations. 
The difference between the source set and the compilations configurations is that the compilation's configurations are transitive.

Reply:
Yup, there is a clear distinction between source set and compilation configurations. However, I'm not sure about the diff between compilation and `classpath` configurations. We can probably chat about that offline.

Reply:
Offline discussion result: 
KGP could, theoretically, also use a 'two layer' approach: 
`source set dependency scope configurations` -> `compile / runtime classpath`

However, the current 'three layer' approach is convenient
`source set dependency scope configurations` -> `fat compilation dependency scope configurations` -> `compile / runtime classpath`

There might be costs to this extra layer, which are currently not measured. 
Changing this architecture, however, is outside the scope of this MR.